### DISCUSSION TITLE
0063号: 記事を1個流しで公開する方法を試したい

### DISCUSSION THREAD

**Discussion by kakutani** [rubima/magazine.rubyist.net#434] [要望・アイデア]
#428 から切り出します( コメントありがとうございます @hsbt )

## リリース方法(の提案)
(これは個別にDiscussionにしたほうがいいのかな?)

旧来の方式——リリース日までに公開Readyな記事を集めて、巻頭言・編集後記などの体裁を整えて「リリース」をアナウンスする、というかたちではなく、Continuous Delivery(CD)にしたいです。

なぜなら:

* リリース日めがけて、複数の記事を「完成」させていくのは作業量や担当者の負荷も高いため、できるだけ「1個流し」にしたい
* るびまの「号」としての価値は、後日参照する(何年も経ってからだったりする)ときに、便利な概念だと思います

だからです。これを踏まえると:

* 「CDにする」とは具体的には、1記事ずつ「完成」したらmasterにマージして、記事単体で「公開」とする
* ある時期がきたら、リリースタグのようなかたちで巻頭言と編集後記をmasterにマージして、それを特定の号のリリースとする
  * 各方面には、このときに「ある号が出た」とアナウンスする
  * ある号に含まれる記事は、その前の号から今回の号までに「公開」された記事とする

**Comment by kakutani** [comment]
>リリース方法(の提案)

基本よさそう、って感じました。一つ気になるのは tag を打つ基準でした。高橋さんの巻頭言が欲しくなったら打つ、みたいな感じ...? となると欲しくなるのはいつ、みたいなループになりそう。

* 四半期とか、半期ごと
* 記事が5つとか8つ溜まったら

とかその辺を合意しておくといいかも。

_Originally posted by @hsbt in https://github.com/rubima/magazine.rubyist.net/issues/428#issuecomment-1554078747_

**Reply by kakutani** [reply]
( #428 からコピペ)
* マージされた記事の様子をうかがって、「そろそろ欲しいな…」みたいな時期にお願いするとよさそう
* めちゃくちゃ雑なんですが RubyKaigi と RubyWorld Conf の後、とかにしてしまうとやっていきの高まり具合でやりやすいのかもしれない。

**Reply by takkanm** [reply]
> 基本よさそう、って感じました。一つ気になるのは tag を打つ基準でした。高橋さんの巻頭言が欲しくなったら打つ、みたいな感じ...? となると欲しくなるのはいつ、みたいなループになりそう。
> 
> * 四半期とか、半期ごと
> * 記事が5つとか8つ溜まったら

なんというか、「今出すべき！！」みたいなテンションのときがあると思うので、リリース後 4 半期以内にだす Kaiju timmer 方式とか(これはこれで記事が貯まるか不安)、Rubyist 節句(RubyKaigi, RubyWorld, リリースみたいなの)にあわせるとかですかね。

**Reply by kakutani** [reply]
「このへんで出せるとうれしいカレンダー」(るびま歳時記)的なものを用意するのはよさそう

**Comment by kakutani** [comment]
「いまはどの記事をデリバリーしようとしているか」がわかるとよさそう(そこでカンバンですよ!)

**Comment by neko314** [comment]
流し方式しかやったことがないので元と比べてどうかと言うのは述べられないのですが、今の所このやり方やりやすいなーと感じています。
0063、 0064号とやってみての反省が2点あり、巻頭言の依頼が遅くなってしまいやすかったり（これは言い訳かもしれない）、「そろそろ号として出せそうかなー」と言うタイミングで新しい記事を書いてくださった人がいる場合にその号に入れるかどうか不明瞭なので迷わせてしまったりしたなーと思っています。

なので、号としては早いうちに締め日・リリース日を決めておくのが良いかと思いました。

「0064号の内容は2024/09/30までに公開した記事とする、0064号のリリース日は2024/10/15とする。」のようなかんじ。

0065号はそうやってみようかなーと思っています🙂
